drivers/ethernet/Kconfig.stm32_hal

```diff
@@ -152,4 +152,11 @@ config ETH_STM32_HAL_PTP_CLOCK_INIT_PRIO
 
 endif # SOC_SERIES_STM32F7X || SOC_SERIES_STM32H7X
 
+config ETH_STM32_USE_HAL_DRIVER_WITH_NEW_API
+	bool "Use new HAL driver"
+	default no
```

Comment:
That is not needed, please remove it. Please see https://docs.zephyrproject.org/1.14.0/guides/kconfig/index.html#redundant-defaults